### PR TITLE
Add eido/convert module

### LINKED ISSUE #2016: new module: EIDO/CONVERT
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by rafalstepien**
## PR checklist

This PR adds EIDO/VALIDATE command that will be used for integrating PEP format with `nf-core` pipelines. Changes:

- Introduced EIDO/CONVERT as a module
- Added unittests

Closes #2016 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/eido/convert/main.nf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://containers.biocontainers.pro/s3/SingImgsRepo/eido/0.1.9_cv1/eido_0.1.9_cv1.sif' :
+        'biocontainers/eido:0.1.9_cv1' }"
+
+    input:
+    path samplesheet
+
+    output:
+    path "versions.yml"           , emit: versions
+    path "*.csv"                  , emit: samplesheet_validated
```
Comment (jfy133) [inline modules/eido/convert/main.nf:15]:
```suggestion
    path "*.csv"                  , emit: samplesheet_converted
```

File: modules/eido/convert/main.nf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    eido convert $samplesheet $args
```
Comment (jfy133) [inline modules/eido/convert/main.nf:23]:
```suggestion
    eido \\
        convert \\
        $samplesheet \\
        $args
```

File: tests/modules/eido/convert/nextflow.config [hunk 3]
```diff
@@ -0,0 +1,7 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+    withName: 'EIDO_CONVERT' {
+        ext.args = '--st-index sample -f csv -p samples=samplesheet_validated.csv'
```
Comment (jfy133) [inline tests/modules/eido/convert/nextflow.config:5]: What is `-p` standing for here? Is that meant to be the output file name? If so this _should_ be moved into the module in this case (and ${prefix}.csv), as in `validate`.

For flexibility and security (to ensure only one CSV is produced), one could alos require the format as an input channel `val(format)` which is then passed to `-f ${format}`, the suffix of `-p ${prefix}.${fornat}` and also thereotially the output glob in the output channel `path(${prefix}.${csv}), emit: output`.

Alternatively, you will need to have one output channel per converstion type output, but made `optional: true, emit: output`

Comment (jfy133) [reply tests/modules/eido/convert/nextflow.config:5]: We can fix this to `csv` in the pipeline-template to make sure it 'works', but some people mayeb use the module outside of nf-core pipelines, so it's good to have it flexible

**Review by jfy133 — CHANGES_REQUESTED** [review]

File: tests/modules/eido/convert/nextflow.config (hunk 3 above)
Comment (rafalstepien) [reply tests/modules/eido/convert/nextflow.config:5]: Yes, the -p stands for paths and is supposed to provide output path (this is kind of confusing but that's not mine implementation). I will try to implement your suggestion.

**Review by jfy133 — APPROVED** [review]
LGTM!

**Pull request merged**